api/src/main/java/org/openmrs/module/fhir/api/util/FHIRConditionUtil.java

```diff
@@ -65,11 +81,243 @@
+		}
+
+		if (!StringUtils.isEmpty(openMrsObs.getComment())) {
+			List<Annotation> annotations = new ArrayList<Annotation>();
+			Annotation annotation = new Annotation();
+			annotation.setText(openMrsObs.getComment());
+			fhirCondition.setNote(annotations);
+		}
 
 		return fhirCondition;
 	}
 
-	public static org.openmrs.Condition generateOpenMRSCondition() {
-		return null;
+	/**
+	 *  Generate Openmrs Obs from FHIR Condition
+	 * @param condition
+	 * @return Obs
+	 */
+
+	public static Obs generateOpenMrsObsFromFHIRCondition(org.hl7.fhir.dstu3.model.Condition condition) {
+		Obs obs = new Obs();
+		List<String> errors = new ArrayList<>();
+		 BaseOpenMRSDataUtil.readBaseExtensionFields(obs, condition);
+
+		if (StringUtils.isNotBlank(condition.getId())) {
+			obs.setUuid(FHIRUtils.extractUuid(condition.getId()));
+		}
+
+		// set concept
+		if (condition.getCode().getText() != null) {
+			Concept concept = Context.getConceptService().getConceptByName(condition.getCode().getText());
+			obs.setConcept(concept);
+		}
+
+		// set person
+		if (condition.getSubject() != null) {
+			Reference subjectRef = condition.getSubject();
+			String patientUuid = subjectRef.getId();
+			Person person = Context.getPersonService().getPersonByUuid(patientUuid);
+			if (person == null) {
+				errors.add("There is no person for the given uuid");
+			} else {
+				obs.setPerson(person);
+			}
+		} else {
+			errors.add("Subject cannot be empty");
+		}
+
+		Date date = condition.getAssertedDate() == null ?  new Date() : condition.getAssertedDate();
+		obs.setDateCreated(date);
+		obs.setObsDatetime(date);
+		obs.setValueText("yes");
+
+		// set comment
+		List<String> notes = new ArrayList<>();
+		for (Annotation note : condition.getNote()) {
+			notes.add(note.getText());
+		}
+		obs.setComment(stringJoin(",", notes));
+
+		return obs;
+	}
+
+	/**
+	 *  Build and set patient reference
+	 * @param omrsEncounter
+	 * @return Reference for the patient
+	 */
+	public static Reference buildPatientReference(org.openmrs.Encounter omrsEncounter) {
+		Reference patientReference = new Reference();
+		PersonName name = omrsEncounter.getPatient().getPersonName();
+		StringBuilder nameDisplay = new StringBuilder();
+		nameDisplay.append(name.getGivenName());
+		nameDisplay.append(" ");
+		nameDisplay.append(name.getFamilyName());
+		String patientUri;
+		nameDisplay.append("(");
+		nameDisplay.append(FHIRConstants.IDENTIFIER);
+		nameDisplay.append(":");
+		nameDisplay.append(omrsEncounter.getPatient().getPatientIdentifier().getIdentifier());
+		nameDisplay.append(")");
+		patientUri = FHIRConstants.PATIENT + "/" + omrsEncounter.getPatient().getUuid();
+		patientReference.setReference(patientUri);
+		patientReference.setDisplay(nameDisplay.toString());
+		return patientReference;
 	}
+
+	/**
+	 *  Joins string in a list
+	 * @param separator
+	 * @param input
+	 * @return string of list items separated by separator specified in the param (commas)
```

Comment:
Doesn't Java already have a standard way of doing this without us having to create and maintain a custom method?